.github/workflows/test-e2e.yml

```
@@ -0,0 +1,272 @@
+              exit 1
+            fi
+            echo "Package resolution failed on attempt $attempt, retrying..."
+            sleep $((attempt * 5))
+          done
+
+      - name: Run UI tests
+        id: tests
+        env:
+          TEST_FILTER: ${{ inputs.test_filter }}
+          TEST_TIMEOUT: ${{ inputs.test_timeout || '120' }}
+        run: |
```

actionlint (1.7.11)

[error] shellcheck reported issue in this script: SC2086:info:12:3: Double quote to prevent globbing and word splitting

(shellcheck)